test/factories/item.factory.ts

```diff
@@ -0,0 +1,38 @@
+import { faker } from '@faker-js/faker';
+
+import { CCLicenseAdaptions, ItemType } from '@graasp/sdk';
+
+import { Item } from '../../src/services/item/entities/Item';
+
+/**
+ * This factory does not guarantee valid items given their type. But they are acceptable when seed into the db.
+ * Create by default a folder
+ * @param item partial properties
+ * @returns complete item valid to be fed into the database
+ */
+export const ItemFactory = (item: Partial<Item>) => ({
+  type: ItemType.FOLDER,
+  order: null,
+  name: faker.word.words(2),
+  description: item.description ?? faker.lorem.text(),
+  extra: { folder: {} },
+  settings:
+    item.settings ??
+    faker.helpers.arrayElement([
+      {},
+      {
+        isPinned: faker.datatype.boolean(),
+        showChatbox: faker.datatype.boolean(),
+        hasThumbnail: false,
+        isResizable: faker.datatype.boolean(),
+        isCollapsible: faker.datatype.boolean(),
+        enableSaveActions: faker.datatype.boolean(),
+        tags: faker.lorem.words({ min: 1, max: 10 }).split(' '),
+        displayCoEditors: faker.datatype.boolean(),
+        ccLicenseAdaption: faker.helpers.enumValue(CCLicenseAdaptions),
+      },
+    ]),
```

Comment:
This means you have to provide a full setting, otherwise you will get ill-defined settings ? should we allow to provide a partial setting that can be filled with default values ?

Reply:
I would rather say that it forces you to set your necessary settings during tests. So if the default changes, it won't break your test. As we discussed I think both approaches are correct, depends on what you want to achieve. I personally prefer a bit of chaos.